mne/time_frequency/tests/test_psd.py

```diff
@@ -13,102 +15,193 @@
 
 
 @requires_version('scipy', '0.12')
-def test_psd():
+def test_psd_deprecate():
     """Test PSD estimation
     """
-    raw = io.Raw(raw_fname)
+    with warnings.catch_warnings(record=True):
+        warnings.simplefilter('always')
+        raw = io.Raw(raw_fname)
 
-    exclude = raw.info['bads'] + ['MEG 2443', 'EEG 053']  # bads + 2 more
+        exclude = raw.info['bads'] + ['MEG 2443', 'EEG 053']  # bads + 2 more
```

Comment:
we typically tend to put the block of `warnings.catch_warnings(record=True)` only immediately before the statement which throws the warning. That way, you know exactly what is creating the warning and also you don't create a huge diff. Also, at the end, you would count if the expected number of warnings were raised. Kind of like here: https://github.com/mne-tools/mne-python/blob/0e660a905e4b2d74d77f9a0fba1a90af8e59e660/mne/tests/test_proj.py#L163